### PR TITLE
ARROW-16408: [C++] Add support for DATE type in SQLite FlightSQL example

### PR DESCRIPTION
**Pull request by drabastomek**
While it works in the integration tests, some SQLite databases have been created (erroneously) with `DATE` data type (instead of `TEXT`). This minor fix handle this edge case.

### CODE REVIEW

**Comment by lidavidm** [comment]
Thanks! Looks like clang-format wants some changes:

```diff
--- /arrow/cpp/src/arrow/flight/sql/example/sqlite_server.cc
+++ /arrow/cpp/src/arrow/flight/sql/example/sqlite_server.cc (after clang format)
@@ -206,8 +206,7 @@
     return float64();
   } else if (boost::iequals(sqlite_type, "BLOB")) {
     return binary();
-  } else if (boost::iequals(sqlite_type, "TEXT") ||
-             boost::iequals(sqlite_type, "DATE") ||
+  } else if (boost::iequals(sqlite_type, "TEXT") || boost::iequals(sqlite_type, "DATE") ||
              boost::istarts_with(sqlite_type, "char") ||
              boost::istarts_with(sqlite_type, "varchar")) {
     return utf8();
@@ -228,8 +227,7 @@
     return SQLITE_FLOAT;
   } else if (boost::iequals(sqlite_type, "BLOB")) {
     return SQLITE_BLOB;
-  } else if (boost::iequals(sqlite_type, "TEXT") ||
-             boost::iequals(sqlite_type, "DATE") ||
+  } else if (boost::iequals(sqlite_type, "TEXT") || boost::iequals(sqlite_type, "DATE") ||
              boost::istarts_with(sqlite_type, "char") ||
              boost::istarts_with(sqlite_type, "varchar")) {
     return SQLITE_TEXT;
```

**Review by lidavidm — APPROVED** [review]
